sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample13_AuthenticateWithClientSecretCredential.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Identity;
+using Azure.Messaging.EventHubs.Samples.Infrastructure;
+
+namespace Azure.Messaging.EventHubs.Samples
+{
+    public class Sample13_AuthenticateWithClientSecretCredential : IEventHubsIdentitySample
+    {
+        /// <summary>
+        ///   The name of the sample.
+        /// </summary>
+        ///
+        public string Name { get; } = nameof(Sample13_AuthenticateWithClientSecretCredential);
+
+        /// <summary>
+        ///   A short description of the sample.
+        /// </summary>
+        ///
+        public string Description { get; } = "An example of authentication using ClientSecretCredential.";
+
+        /// <summary>
+        ///   Runs the sample using the specified Event Hubs connection information.
+        /// </summary>
+        ///
+        /// <param name="fullyQualifiedNamespace">The fully qualified Event Hubs namespace.  This is likely to be similar to <c>{yournamespace}.servicebus.windows.net</param>
+        /// <param name="eventHubName">The name of the Event Hub, sometimes known as its path, that the sample should run against</param>
+        /// <param name="tenantId">The Azure Active Directory tenant that holds the service principal</param>
+        /// <param name="clientId">The Azure Active Directory client identifier of the service principal</param>
+        /// <param name="secret">The Azure Active Directory secret of the service principal</param>
+        ///
+        public async Task RunAsync(string fullyQualifiedNamespace,
+                                   string eventHubName,
+                                   string tenantId,
+                                   string clientId,
+                                   string secret)
+        {
+            // Service principal authentication is a mean for applications to authenticate
+            // against Azure Active Directory (AAD) and consume Azure services. This is advantageous compared
+            // to signing in using fully privileged users as it allows to enforce role-based authorization
+            // from the portal.
+            //
+            // Service principal authentication differs from managed identity authentication also because the principal to be used
+            // will be able to authenticate with the portal without the need for the code to run within the portal.
+            // The authentication between the Event Hubs client and the portal is performed through OAuth 2.0.
+            // (see: https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals)
+
+            // ClientSecretCredential allows performing service principal authentication passing
+            // tenantId, clientId and clientSecret directly from the constructor.
+            // (see: https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-client-creds-grant-flow)
```

Comment:
Nicely done!    This is very clear and well-written.   Thank you for your great attention to detail here.